recipes-bsp/u-boot/files/0006-dts-iot2050-Support-new-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,634 @@
++		compatible = "everlight,pm16d17";
++		reg = <0x44>;
++
++		ps-gain = <1>;
++		ps-itime = "0.4";
++		ps-wtime = "25";
++		ps-ir-led-pulse-count = <1>;
++	};
++
++	/delete-node/ edp-bridge@f;
++};
++
++/delete-node/ &pcal9535_1;
++/delete-node/ &pcal9535_2;
++/delete-node/ &pcal9535_3;
++
++&dss {
++	status = "disabled";
++};
++
++&dss_ports {
++	/delete-node/ port@1;
++};
++
++&ecap0 {
++	status = "disabled";
++};
++
++&mcu_uart0 {
++	status = "disabled";
++};
++
++&tscadc1 {
++	status = "disabled";
++};
++
++&serdes0 {
++	assigned-clocks = <&k3_clks 153 4>, <&serdes0 AM654_SERDES_CMU_REFCLK>;
++	assigned-clock-parents = <&k3_clks 153 8>, <&k3_clks 153 4>;
++};
++
++&serdes1 {
++	status = "disabled";
++};
++
++&pcie0_rc {
++	pinctrl-names = "default";
++	pinctrl-0 = <&minipcie_pins_default>;
++
++	num-lanes = <1>;
++	phys = <&serdes0 PHY_TYPE_PCIE 1>;
++	phy-names = "pcie-phy0";
++	reset-gpios = <&wkup_gpio0 27 GPIO_ACTIVE_HIGH>;
++	status = "okay";
++};
++
++&pcie0_ep {
++	status = "disabled";
++};
++
++&pcie1_rc {
++	status = "disabled";
++};
++
++&pcie1_ep {
++	status = "disabled";
++};
++
++&dwc3_0 {
++	assigned-clock-parents = <&k3_clks 151 4>,  /* set REF_CLK to 20MHz i.e. PER0_PLL/48 */
++				 <&k3_clks 151 9>;  /* set PIPE3_TXB_CLK to CLK_12M_RC/256 (for HS only) */
++	/delete-property/ phys;
++	/delete-property/ phy-names;
++};
++
++&usb0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <
++		&usb0_pins_default
++	>;
++
++	maximum-speed = "high-speed";
++	/delete-property/ snps,dis-u1-entry-quirk;
++	/delete-property/ snps,dis-u2-entry-quirk;
++};
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index 0b0686e2628b..d0d8b8d75700 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -38,6 +38,8 @@ struct iot2050_info {
+ 	u8 mac_addr_cnt;
+ 	u8 mac_addr[8][ARP_HLEN];
+ 	char seboot_version[40 + 1];
++	u8 padding[3];
++	u32 ddr_size_mb;
+ } __packed;
+ 
+ /*
+@@ -172,6 +174,14 @@ static bool board_is_m2(void)
+ 		strcmp((char *)info->name, "IOT2050-ADVANCED-M2") == 0;
+ }
+ 
++static bool board_is_sm(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strcmp((char *)info->name, "IOT2050-ADVANCED-SM") == 0;
++}
++
+ static void remove_mmc1_target(void)
+ {
+ 	char *boot_targets = strdup(env_get("boot_targets"));
+@@ -185,6 +195,15 @@ static void remove_mmc1_target(void)
+ 	free(boot_targets);
+ }
+ 
++static void enable_pcie_connector_power(void)
++{
++	if (board_is_sm())
++		set_pinvalue("gpio@601000_22", "P3V3_PCIE_CON_EN", 1);
++	else
++		set_pinvalue("gpio@601000_17", "P3V3_PCIE_CON_EN", 1);
++	udelay(4 * 100);
++}
++
+ void set_board_info_env(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -222,6 +241,8 @@ void set_board_info_env(void)
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced.dtb";
+ 		else if(board_is_m2())
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced-m2.dtb";
++		else if(board_is_sm())
```

Comment:
Style: `if (...` - actually already for the m2 test above.

Reply:
ACK